### PR TITLE
DATAUP-314: job log viewer

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Job log viewer changes, pt I

This is nearly all search-and-replace changes that I made to make navigating the code and understanding its functionality easier: renaming variables and functions, moving bits of code to different places in the file, and so on. The functionality of the widget has not yet been changed.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-314
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative, finding your favourite bulk import cell, and checking out the logs. Whoa!! They look amazing, don't they?

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 1]
```diff
@@ -50,17 +54,17 @@ define([
                     auto: true,
                 },
                 {
-                    mode: 'active',
+                    mode: 'running',
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:57]: changed the mode names to match the job statuses -- makes things a lot easier!

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 2]
```diff
@@ -402,9 +411,11 @@ define([
                     totalLines: null,
                 },
             }),
-            runtime = Runtime.make(),
-            loopFrequency = 5000,
-            listeners = [];
+            bus = Runtime.make().bus(),
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:414]: replaced `runtime.bus()` with `bus` throughout the file

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 3]
```diff
@@ -417,18 +428,21 @@ define([
             requestLoop = null, // the timeout object
             scrollToEndOnNext = false;
 
-        // VIEW ACTIONS
+        function getLogPanel() {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:431]: moved up from lower down in the file

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 4]
```diff
@@ -417,18 +428,21 @@ define([
...
+            return ui.getElement('log-panel');
+        }
 
-        function scheduleNextRequest() {
+        // VIEW ACTIONS
+        function scheduleLogRequest() {
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:436]: changed the method names to include what the request is for -- status or log -- to make life easier for developers

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 5]
```diff
@@ -524,6 +542,37 @@ define([
             requestLatestJobLog();
         }
 
+        /**
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:545]: moved up from lower in the file

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 6]
```diff
@@ -626,139 +674,132 @@ define([
...
+                                        dataToggle: 'collapse',
+                                        dataTarget: '#' + uniqueID,
+                                        ariaExpanded: 'false',
+                                        ariaControls: uniqueID,
+                                    },
+                                    [span({}, ['Logs'])]
+                                ),
+                                div(
+                                    {
+                                        class: `${cssBaseClass}__logs_content_panel collapse`,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:727]: collapsible logs panel

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 7]
```diff
@@ -626,139 +674,132 @@ define([
...
                     },
                     [
-                        renderControls(events),
-                        div({
-                            dataElement: 'log-panel',
-                            class: logContentClass,
-                        }),
+                        div(
+                            {
+                                class: `${cssBaseClass}__status_container`,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:699]: container for the job status lines (will be added in next PR)

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 8]
```diff
@@ -940,15 +955,15 @@ define([
...
             }
             if (looping) {
-                scheduleNextRequest();
+                scheduleLogRequest();
             }
         }
 
         function startEventListeners() {
-            let ev = runtime.bus().listen({
+            listenersByType['job-logs'] = bus.listen({
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:966]: save listeners by key to facilitate removing them when a job has completed

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 9]
```diff
@@ -1019,37 +1030,25 @@ define([
...
-                lineNumber: '',
-                text: 'Job is queued, logs will be available when the job is running.',
-            };
-            const line = buildLine(noLogYet);
-            getLogPanel().appendChild(line);
+        function doOnQueued() {
+            ui.setContent(
+                'log-panel',
+                p(['Job is queued; logs will be available when the job is running.'])
+            );
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:1039]: just add the message directly, don't faff around with pretending it is a log line

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 10]
```diff
@@ -1060,38 +1059,36 @@ define([
...
+            fsm.bus.on('on-terminated', () => {
                 requestLatestJobLog();
-                stopJobUpdates();
-            });
-            fsm.bus.on('exit-canceled', () => {
-                //  nothing to do?
+                stopJobStatusUpdates();
             });
-            fsm.bus.on('on-queued', (message) => {
-                doOnQueued(message);
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:1077]: none of these `message` variables are used

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 11]
```diff
@@ -626,139 +674,132 @@ define([
...
-            wrapperDiv.appendChild(textDiv);
-            // append wrapper to line div
-            kblogLine.appendChild(wrapperDiv);
-
-            return kblogLine;
-        }
-
-        function getLogPanel() {
-            return ui.getElement('log-panel');
+        function renderLogLine(line) {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:759]: So much better!

**Review by briehl — APPROVED** [review]
Yup, looks like mostly some reordering and variable renaming. :+1:

**Pull request merged**